leader-election-impl/src/main/java/com/palantir/leader/PaxosRequestBatcher.java

```diff
@@ -0,0 +1,90 @@
+package com.palantir.leader;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Maps;
+import com.palantir.leader.LeaderElectionService.LeadershipToken;
+import com.palantir.leader.LeaderElectionService.StillLeadingStatus;
+
+public class PaxosRequestBatcher {
+    private static final Logger log = LoggerFactory.getLogger(PaxosRequestBatcher.class);
+
+    private final ConcurrentMap<LeadershipToken, Batch> queue;
+    private final Function<LeadershipToken, StillLeadingStatus> operation;
+
+    public PaxosRequestBatcher(ExecutorService executor, Function<LeadershipToken, StillLeadingStatus> operation) {
+        this.queue = Maps.newConcurrentMap();
+        this.operation = operation;
+        executor.execute(() -> sendRequests());
+    }
+
+    private void sendRequests() {
+        while (true) {
+            if (queue.isEmpty()) {
+                continue;
+            }
+
+            for (Map.Entry<LeadershipToken, Batch> e : queue.entrySet()) {
+                populateBatch(e.getValue(), e.getKey());
+            }
```

Comment:
Do we want to catch Throwable to ensure this method never terminates and kills the task/thread?